### DISCUSSION TITLE
LESSON#7  Why lib/forge-std/lib folder is empty?

### DISCUSSION THREAD

**Discussion by Abdulllah-Rizwan** [Cyfrin/foundry-full-course-cu#550] [Q&A] [answered]
Hey Everyone I initialize foundry framework and got folder structures however initially lib folder was missing which i later installed via this command `git submodule add --force --name lib/forge-std https://github.com/foundry-rs/forge-std lib/forge-std`.Now when i try to run test on my smart contract I came to know that lib/forge-std/lib folder is completely empty whereby my test failed to run how can i get all the files and folders that lies under lib/forge-std/lib ? 

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112887740/b5339cee-cd6f-4578-918d-85208ccff676)

**Comment by Abdulllah-Rizwan** [comment, marked as answer]
okay i found the solution which was a command as `forge install --no-commit` which installed the remaining folders and files and thus my test ran well

**Reply by cromewar** [reply]
Amazing catch @Abdulllah-Rizwan

**Comment by khushbusandhu** [comment]
i was also getting the same problem 
thanks for your help :)

**Comment by 0xKumarShivam** [comment]
this error comes when you don't  set user id and password of git . go and set git user id and passoward
```git config --global user.name "Your Name"```
  ```git config --global user.email "you@example.com"```

do it one by one , but 1st open wsl command prompt (Ubuntu) do it there  for permanent soluiton of this problem.

**Reply by Abdulllah-Rizwan** [reply]
Thank you soo much ❤
